lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -95,17 +96,31 @@ public LockImmutableTimestampResponse lockImmutableTimestamp() {
                     LockTokenConverter.toTokenV2(lock));
         } catch (Throwable e) {
             if (lock != null) {
-                lockService.unlock(lock);
+                try {
+                    lockService.unlock(lock);
+                } catch (Throwable unlockThrowable) {
+                    e.addSuppressed(unlockThrowable);
+                }
             }
             throw Throwables.rewrapAndThrowUncheckedException(e);
         }
     }
 
     @Override
     public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        return StartIdentifiedAtlasDbTransactionResponse.of(
-                lockImmutableTimestamp(),
-                TimestampAndPartition.of(getFreshTimestamp(), 0));
+        LockImmutableTimestampResponse immutableTimestamp = lockImmutableTimestamp();
+        try {
+            return StartIdentifiedAtlasDbTransactionResponse.of(
+                    immutableTimestamp,
+                    TimestampAndPartition.of(getFreshTimestamp(), 0));
+        } catch (RuntimeException | Error throwable) {
+            try {
+                unlock(ImmutableSet.of(immutableTimestamp.getLock()));
+            } catch (Throwable unlockThrowable) {
+                throwable.addSuppressed(unlockThrowable);
+            }
+            throw Throwables.rewrapAndThrowUncheckedException(throwable);
```

Comment:
nit: `throwable` is unchecked, so we can throw it without a rewrap

Reply:
done